var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py

```diff
@@ -140,14 +141,28 @@ class IntelOneapiMpi(IntelOneApiLibraryPackage):
     conflicts("+generic-names +classic-names")
 
     executables = [r"^mpiicpx$"]
-    version_regex = r"Intel\(R\) MPI Library (\S+)"
 
     @classmethod
     def determine_version(cls, exe):
-        output = Executable(exe)("-v", output=str, error=str)
-        match = re.search(cls.version_regex, output)
-        # strip @ from unsubstituted @IMPI_OFFICIALVERSION@
-        return match.group(1).strip("@") if match else None
+        output = Executable(exe)("-show", output=str, error=str)
+        match = re.search(r"/mpi/(20\d\d\.\d+(\.\d+)?)/", output)
+        return match.group(1) if match else None
```

Comment:
I don't think this is the right approach. It parses the directory structure that is installed in, which should be considered an implementation detail. The previous version, `mpiicpx -v`, explicitly queries the version.

Reply:
The previous version works with 2021.14 in ubuntu, so the `IMPI_OFFICIALVERSION` issue is resolved.